### PR TITLE
Init

### PR DESCRIPTION
**Pull request by jonenst**
WIPTODO:
* [x] idx/pk on data table
* [x] col filter

in the future:
* [ ] separate column for row number (read thread slicing) and row time ?
* [ ] row filter by row num
* [ ] row filter by row time

### CODE REVIEW

File: src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+    // TODO tune these parameters for performance
+    // TODO make these parameters in application.yaml
+    private static final int WRITE_BATCHSIZE = 30000;
+    private static final int WRITE_THREADSIZE = 3; // 3 batches => e.g. 300 rows of 300cols
+    private static final int READ_THREADSIZE = 300; // 300 db rows, TODO take the number of cols into account
+
+    public void save(UUID uuid, List<TimeSeries> listTimeseries) {
+        try {
+            dosave(uuid, listTimeseries);
```
Comment (thangqp) [inline src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java:84]: camel case for variables and method => to revisite all dosave and dofindById

File: src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+                "insert {} in batch of {} rows ({} doubles for each batch), numbatch={}, numthreads={}, threadbatches={}",
+                uuid, batchrow, batchrow * colcount, batchcount, threadcount, threadbatches);
+        long a = System.nanoTime();
+
+        // TODO here we transpose, which means it's impossible to stream
+        // data from the client to the database, the server has to buffer in memory.
+        // try to change the API to allow streaming.
+        // TODO avoid copying the data (timeseries toArray())?
+        BiFunction<Integer, Integer, Object> stringOrDoubledataGetter;
+        if (listTimeseries.get(0) instanceof DoubleTimeSeries) {
```
Comment (thangqp) [inline src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java:114]: in metadata, we have info of dataType, is possible avoid check with the first element by get(0)
"metadata" : {
      "name" : "_GEN____1_SM_generator_UStatorPu",
      "dataType" : "DOUBLE",
      "tags" : [ ],
      "irregularIndex" : []
    }

File: src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+            List<double[]> datadouble = new ArrayList<>();
+            for (int i = 0; i < listTimeseries.size(); i++) {
```
Comment (thangqp) [inline src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java:116]: in my opinion, using for-each is less error prone..

File: src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+                // TODO get compress mode from the metadata sent by the client
+                if (tryToCompress) {
+                    ddc = ddc.tryToCompress();
+                }
+                // TODO more types
+                // TODO index from client
+                TimeSeries timeseries = new StringTimeSeries(timeseriesMetadataService.getMetadata(index, individualMetadatas, entry.getKey()), List.of(ddc));
+                ret.add(timeseries);
+            } else {
+                throw new RuntimeException("Unsupported read of timeseries type" + entry.getValue().get(0).getClass());
```
Comment (thangqp) [inline src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java:324]: As above, in my opinion, checking type by 'dataType' is a better choice. I try to test with rest endpoint to get an empty time line, an exception 
java.lang.NullPointerException: null
	at org.gridsuite.timeseries.server.TimeseriesDataRepository.dofindById(TimeseriesDataRepository.java:324) ~[classes/:na]
	at org.gridsuite.timeseries.server.TimeseriesDataRepository.findById(TimeseriesDataRepository.java:200) ~[classes/:na]
	at org.gridsuite.timeseries.server.TimeseriesDataRepository$$FastClassBySpringCGLIB$$1ac395ff.invoke(<generated>) ~[classes/:na]  

An example of empty timeline produce by dynawalz

"timeLine" : {
    "metadata" : {
      "name" : "timeLine",
      "dataType" : "STRING",
      "tags" : [ ],
      "regularIndex" : {
        "startTime" : 0,
        "endTime" : 0,
        "spacing" : 0
      }
    },
    "chunks" : [ ]
  }

File: src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+import com.powsybl.timeseries.TimeSeriesIndex;
+import com.powsybl.timeseries.UncompressedDoubleDataChunk;
+import com.powsybl.timeseries.UncompressedStringDataChunk;
+import com.zaxxer.hikari.HikariDataSource;
+
+/**
+ * @author Jon Schuhmacher <jon.harper at rte-france.com>
+ */
+@Repository
+public class TimeseriesDataRepository {
```
Comment (thangqp) [inline src/main/java/org/gridsuite/timeseries/server/TimeseriesDataRepository.java]: In the future, in my opinion, we can extract all technical codes which uses multithreads into an under layer, e.g. a DAO class which processes in batch mode for CRUD operations.

**Review by thangqp — APPROVED** [review]
I send this timeLine
[
	{
	  "metadata" : {
		"name" : "timeLine",
		"dataType" : "STRING",
		"tags" : [ ],
		"regularIndex" : {
		  "startTime" : 0,
		  "endTime" : 0,
		  "spacing" : 0
		}
	  },
	  "chunks" : [ ]
	}
]
but when testing with swagger, I receive
[
  {
    "metadata": {
      "name": "timeLine",
      "dataType": "STRING",
      "tags": [],
      "regularIndex": {
        "startTime": 0,
        "endTime": 0,
        "spacing": 0
      }
    },
    "chunks": [
      {
        "offset": 0,
        "values": [
          null
        ]
      }
    ]
  }
]

File: src/main/java/org/gridsuite/timeseries/server/TimeSeriesDataRepository.java [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+public class TimeSeriesDataRepository {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TimeSeriesDataRepository.class);
+
+    private static final String INSERT = "insert into timeseries_group_data ( group_id, time, json_obj ) values (?,?,?);";
+    private static final String COUNT = "select count(*) from timeseries_group_data where group_id=?;";
+    private static final String SELECTALL = "select time, json_obj from timeseries_group_data where group_id=? and time>=? and time <? order by time;";
+    private static final String DELETE = "delete from timeseries_group_data where group_id=?";
+
+    private String makeSelectAll(List<String> col) {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/timeseries/server/TimeSeriesDataRepository.java:53]: Maybe we should move all query buildling on a custom catalog class.

**Pull request merged**